test/java/seedu/addressbook/commands/ViewAllCommandTest.java

```diff
@@ -0,0 +1,150 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.util.TestUtil;
+
+public class ViewAllCommandTest {
+    private AddressBook addressBook;
+    private AddressBook emptyAddressBook;
+    private List<ReadOnlyPerson> emptyDisplayList;
+    private List<ReadOnlyPerson> listWithAll;
+    private List<ReadOnlyPerson> listWithSome;
+
+    @Before
+    public void setUp() throws Exception {
+        Person johnDoe = new Person(new Name("John Doe"),
+                                    new Phone("98765432", false),
+                                    new Email("johnd@gmail.com", false),
+                                    new Address("John street, block 123, #01-01", false),
+                                    new UniqueTagList(Collections.emptySet()));
+        Person betsyCrowe = new Person(new Name("Betsy Crowe"),
+                                       new Phone("1234567", true),
+                                       new Email("betsycrowe@gmail.com", false),
+                                       new Address("Newgate Prison", true),
+                                       new UniqueTagList(new Tag("friend"), new Tag("criminal")));
+
+        emptyAddressBook = TestUtil.createAddressBook();
+        addressBook = TestUtil.createAddressBook(johnDoe, betsyCrowe);
+
+        emptyDisplayList = TestUtil.createList();
+        listWithAll = TestUtil.createList(johnDoe, betsyCrowe);
+        listWithSome = TestUtil.createList(betsyCrowe);
+    }
+
+    @Test
+    public void viewAllCommand_invalidIndex_returnsInvalidIndexMessage() {
+        // empty addressbook
+        assertViewAllErrorInvalidIndex(emptyAddressBook, emptyDisplayList, 1);
+
+        // non-empty addressbook
+        assertViewAllErrorInvalidIndex(addressBook, listWithAll, -1);
+        assertViewAllErrorInvalidIndex(addressBook, listWithAll, 0);
+        assertViewAllErrorInvalidIndex(addressBook, listWithAll, 3);
```

Comment:
if you use `size()+1` of the list instead of magic number `3`, the test will be less brittle (less likely to break in future).